core/src/main/java/com/linecorp/armeria/client/EventLoopScheduler.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Streams;
+
+import com.linecorp.armeria.common.util.Releasable;
+
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+
+final class EventLoopScheduler {
+
+    private final EventLoop[] eventLoops;
+    private final Map<String, State> map = new ConcurrentHashMap<>();
+    private int counter;
+    private volatile long lastCleanupTimeNanos = System.nanoTime();
+
+    EventLoopScheduler(EventLoopGroup eventLoopGroup) {
+        eventLoops = Streams.stream(eventLoopGroup)
+                            .map(e -> (EventLoop) e)
+                            .toArray(EventLoop[]::new);
+    }
+
+    Entry acquire(Endpoint endpoint) {
+        requireNonNull(endpoint, "endpoint");
+        final State state = state(endpoint);
+        final Entry acquired = state.acquire();
+        cleanup();
+        return acquired;
+    }
+
+    @VisibleForTesting
+    List<Entry> entries(Endpoint endpoint) {
+        return state(endpoint).entries();
+    }
+
+    private State state(Endpoint endpoint) {
+        final String authority = endpoint.authority();
+        return map.computeIfAbsent(authority, e -> new State(eventLoops));
+    }
+
+    /**
+     * Cleans up empty entries with no activity for more than 1 minute. For reduced overhead, we perform this
+     * only when 1) the last clean-up was more than 1 minute ago and 2) the number of acquisitions % 256 is 0.
+     */
+    private void cleanup() {
+        if ((++counter & 0xFF) != 0) {
```

Comment:
This is fine though I'd expect at runtime it's equivalent to `% 256` which would be more readable.